app/javascript/packs/validateFileInputs.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * @file Manages logic to validate file uploads client-side. In general, the
+ * validations work by looping over input form fields with a type of file and
+ * checking the size and format of the files upload by the user.
+ */
+
+/**
+ * An object containing the top level MIME type as the key and the max file
+ * size in MB for the value. To use a different value than these defaults,
+ * simply add a data-max-file-mb attribute to the input form field with the
+ * max file size in MB. If that attribute is found, it takes priority over these
+ * defaults.
+ *
+ * @constant {Object.<string, number>}
+ */
+const MAX_FILE_SIZE_MB = Object.freeze({
+  image: 25,
+  video: 50,
+});
+
+/**
+ * Permitted file types using the top level MIME type (i.e. image for
+ * image/png). To specify permitted file types, simply add a
+ * data-permitted-file-types attribute to the input form field as an Array of
+ * strings specifying the top level MIME types that are permitted.
+ *
+ * @constant {string[]}
+ */
+const PERMITTED_FILE_TYPES = ['video', 'image'];
+
+/**
+ * Removes any pre-existing error messages from the DOM related to file
+ * validation.
+ *
+ * @param {HTMLElement} fileInput - An input form field with type of file
+ */
+function removeErrorMessages(fileInput) {
+  const errorMessages = fileInput.parentNode.querySelectorAll(
```

Comment:
How about wrapping all error messages in a single div, then selecting that one element here and iterating on `childNodes`?

```javascript
const errorMessages = fileInput.parentNode.querySelector('.file-upload-errors')
// Need to turn NodeList into array
[...errorMessages.childNodes].forEach... 
```

Reply:
That's a good idea!

I think it's inspiring some further cleanup. There's not a situation where we can have 2 error messages on the same field because I'm breaking from the loop as soon as I find the first error. So if a file is too big AND the wrong type, it throws the file too big error because that's checked first (intentional).

I noticed I have `addErrorMessage` (singular) and `removeErrorMessages` (plural). I should sync that up to be more clear.

Great catch!